File: test/lira_output/ts/notes_app/memory_store.ts
// module notes_memory_store
import { Note } from "./models";

import { NoteStore } from "./store";

export class MemoryNoteStore implements NoteStore {
  private notes: Note[] = [];
  public get(id: string): Note | null {
    for (const note of this.notes) {
      if ((note.id == id)) {
        return note;
      }
    }
    return null;
  }
  public save(note: Note): Note {
    this.notes.push(note);
    return note;
  }
  public list(): Note[] {
    return this.notes;
  }
}

File: test/lira_output/py/notes_app/memory_store.py
# module notes_memory_store
from __future__ import annotations
from .models import Note

from .store import NoteStore

class MemoryNoteStore(NoteStore):
    notes: list[Note] = []

    def get(self, id: str) -> Note | None:
        for note in self.notes:
            if (note.id == id):
                return note
        return None


    def save(self, note: Note) -> Note:
        self.notes.append(note)
        return note


    def list(self) -> list[Note]:
        return self.notes
